### PR TITLE
Grouping static checks under the same task

### PR DESCRIPTION
**Pull request by nhachicha**
Java-5633

### CODE REVIEW

File: .evergreen/run-kotlin-tests.sh [hunk 1]
```diff
@@ -34,4 +34,4 @@ fi
 echo "Running Kotlin tests"
 
 ./gradlew -version
-./gradlew kotlinCheck -Dorg.mongodb.test.uri=${MONGODB_URI} ${MULTI_MONGOS_URI_SYSTEM_PROPERTY}
```
Comment (rozza) [inline .evergreen/run-kotlin-tests.sh:37]: Ah!this is probably why `kotlinCheck` was added as a task. Here nothing will run as there is no task passed to `gradlew`.  

A couple of options could be:

1) Keep `kotlinCheck` and only depend on test tasks
2) List all the kotlin projects and the tasks you want to run:

```
./gradlew bson-kotlin:test bson-kotlinx:test driver-kotlin-sync:test driver-kotlin-sync:integrationTest ...
```

Comment (nhachicha) [reply .evergreen/run-kotlin-tests.sh:37]: I think 2 is more explicit/readable about what we want to test 👍

File: bson-kotlin/build.gradle.kts [hunk 2]
```diff
@@ -111,12 +111,11 @@ spotbugs { showProgress.set(true) }
 // ===========================
 //     Test Configuration
 // ===========================
-tasks.create("kotlinCheck") {
-    description = "Runs all the kotlin checks"
+tasks.findByName("check")?.run {
+    description = "Runs static checks and unit test"
```
[the remark this answers is not in the document]
Comment (nhachicha) [reply bson-kotlin/build.gradle.kts:115]: Yes, this is why they were previously excluded in https://github.com/mongodb/mongo-java-driver/pull/1526/files#diff-0df78703287dbf5248e8a9ce9cb218d7290e723227854cb9e0a723cbfc8b2d41R15 (`-x test`)

[the remark this answers is not in the document]
Comment (jyemin) [reply bson-kotlin/build.gradle.kts:115]: Got it, thanks.

File: bson-kotlin/build.gradle.kts [hunk 3]
```diff
@@ -111,12 +111,11 @@ spotbugs { showProgress.set(true) }
 // ===========================
 //     Test Configuration
 // ===========================
-tasks.create("kotlinCheck") {
```
Comment (rozza) [inline bson-kotlin/build.gradle.kts:114]: I think these can be deleted. Running `--dry-run` with gradle shows you the tasks to be run:

```
./gradlew bson-kotlin:check --dry-run | grep "bson-kotlin"
:bson-kotlin:compileKotlin SKIPPED
:bson-kotlin:compileJava SKIPPED
:bson-kotlin:compileGroovy SKIPPED
:bson-kotlin:processResources SKIPPED
:bson-kotlin:classes SKIPPED
:bson-kotlin:jar SKIPPED
:bson-kotlin:inspectClassesForKotlinIC SKIPPED
:bson-kotlin:checkstyleMain SKIPPED
:bson-kotlinx:compileKotlin SKIPPED
:bson-kotlinx:compileJava SKIPPED
:bson-kotlinx:compileGroovy SKIPPED
:bson-kotlinx:processResources SKIPPED
:bson-kotlinx:classes SKIPPED
:bson-kotlinx:jar SKIPPED
:bson-kotlinx:inspectClassesForKotlinIC SKIPPED
:bson-kotlin:compileTestKotlin SKIPPED
:bson-kotlin:compileTestJava SKIPPED
:bson-kotlin:compileTestGroovy SKIPPED
:bson-kotlin:processTestResources SKIPPED
:bson-kotlin:testClasses SKIPPED
:bson-kotlin:checkstyleTest SKIPPED
:bson-kotlin:codenarcMain SKIPPED
:bson-kotlin:codenarcTest SKIPPED
:bson-kotlin:detekt SKIPPED
:bson-kotlin:spotbugsMain SKIPPED
:bson-kotlin:spotbugsTest SKIPPED
:bson-kotlin:spotlessExtraneous SKIPPED
:bson-kotlin:spotlessExtraneousApply SKIPPED
:bson-kotlin:spotlessKotlin SKIPPED
:bson-kotlin:spotlessKotlinApply SKIPPED
:bson-kotlin:spotlessKotlinGradle SKIPPED
:bson-kotlin:spotlessKotlinGradleApply SKIPPED
:bson-kotlin:spotlessApply SKIPPED
:bson-kotlin:spotlessExtraneousCheck SKIPPED
:bson-kotlin:spotlessKotlinCheck SKIPPED
:bson-kotlin:spotlessKotlinGradleCheck SKIPPED
:bson-kotlin:spotlessCheck SKIPPED
:bson-kotlin:test SKIPPED
:bson-kotlin:check SKIPPED
```

**Review by rozza — CHANGES_REQUESTED** [review]
One last recommended change then LGTM

File: bson-kotlin/build.gradle.kts (hunk 3 above)
Comment (nhachicha) [reply bson-kotlin/build.gradle.kts:114]: Good point 👍

**Review by rozza — APPROVED** [review]
LGTM!

**Pull request merged**